templates/codepipeline-cf-template-codecommit-source.yml

```diff
@@ -31,17 +31,23 @@ Parameters:
     Description: Environment
     Type: String
     Default: dev
+  
+  ParameterName:
+    Description: The parameter name you stored in ssm for your auth token
+    Type: String
+    Default: auth_token
```

Comment:
So, what should we specify here? I understand `auth_token` is a default value, but if that's left as is then I don't think it will work. The SSM permissions require the full resource name, e.g. `/MozDef-EF/zoom/dev/auth_token`, instead of just `auth_token`.

Or, we don't need this as a parameter, and instead apply the SSM permissions on a more specific resource, as suggested by Gene in his recommendation: `!Join [ '', [ 'arn:aws:ssm:', !Ref 'AWS::Region', ':', !Ref 'AWS::AccountId', ':parameter/MozDef-EF/', !Ref Service, '/', !Ref Environment, '/auth_token' ] ]`.
(so the above value instead of line 252 in this commit)

I am of the opinion of having as little CF template parameters as possible, to reduce complexity. I think, once a number of parameter/environment variables are decided upon, having as many resources parametrised based on those values are less error-prone than leaving it to the user to enter manually.

What do you think?

Reply:
I've already tested it, it works

Reply:
actually you're right on that, it should be the full parameter as I look back at where this is used:
            Resource: !Join [ '', [ 'arn:aws:ssm:', !Ref 'AWS::Region', ':', !Ref 'AWS::AccountId', ':parameter/', !Ref 'ParameterName' ] ]

Reply:
I am of the opinion of not using the "ParameterName" CF parameter, and use this for the resource line: `!Join [ '', [ 'arn:aws:ssm:', !Ref 'AWS::Region', ':', !Ref 'AWS::AccountId', ':parameter/', !Ref ProjectName, '/', !Ref Service, '/', !Ref Environment, '/auth_token' ] ]`